recipes/libcufile/meta.yaml

```
@@ -0,0 +1,160 @@
+  sha256: 222c49b29afb544c69542c4e658b0a1bd01fd2306f2136ad6990ccfff482a097  # [linux64]
+  sha256: 4e8d2b525322be9ab6ae574bd1d0325a4bf9c6c3bbdaa1611b5b969b7b34b0c4  # [aarch64]
+  patches:
+    - Makefile.patch
+
+build:
+  number: 0
+  skip: true  # [not linux64]
+
+outputs:
+  - name: libcufile
+    build:
+      missing_dso_whitelist:
+        - "*/libcuda.so.*"                     # [linux]
+      run_exports:
+        - {{ pin_subpackage("libcufile", max_pin="x") }}
+    files:
+      - lib/libcufile*.so.*
+      - etc/cufile.json
+      - man
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17   # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcufile.so.*
+        - test -f $PREFIX/lib/libcufile_rdma.so.*
+        - test -f $PREFIX/etc/cufile.json
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_file: LICENSE
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: |
+        Library for GPU Direct Storage with CUDA
+      doc_url: https://docs.nvidia.com/gpudirect-storage/api-reference-guide/index.html
+
+  - name: libcufile-dev
+    build:
+      run_exports:
+        - {{ pin_subpackage("libcufile", max_pin="x") }}
+    files:
+      - lib/libcufile*.so
+      - targets/*/lib/libcufile*.so
+      - include/cufile*
+      - targets/*/include/cufile*
+    requirements:
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - libcufile >={{ version }}
+      run_constrained:
+        - libcufile-static >={{ version }}
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcufile.so
+        - test -f $PREFIX/lib/libcufile_rdma.so
+        - test -f $PREFIX/include/cufile*
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_file: LICENSE
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: |
+        Library for GPU Direct Storage with CUDA
+      doc_url: https://docs.nvidia.com/gpudirect-storage/api-reference-guide/index.html
+
+  - name: libcufile-static
+    files:
+      - lib/libcufile*_static.a
+      - targets/*/lib/libcufile*_static.a  
+    requirements:
+      host:
+        - cuda-version {{ cuda_version }}
+        - libcufile-dev {{ version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - {{ pin_subpackage("libcufile", max_pin="x") }}
```

We're not including dependencies between `-dev` and `-static` except for a `run_constrained` on the `-dev` package to ensure the static library version is `>=` the dev package version.
```suggestion
      host:
        - cuda-version {{ cuda_version }}
      run:
        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
```